Documentation/API.md

```diff
@@ -214,6 +199,61 @@ public enum MyEnum: SwiftProtobuf.Enum {
 }
 ```
 
+### Enum and enum case naming
+
+The name of the Swift enum is copied directly from the name in the proto file,
+prefixed with the package name or the name from `option swift_prefix`
+as documented above for messages.
+If that name would conflict with a Swift reserved word or otherwise
+cause problems for the generated code, the word `Enum` will
+be appended to the name.
+
+Enum case names are converted from `UPPER_SNAKE_CASE` conventions
+in the proto file to `lowerCamelCase` in the Swift code.
+
+If the enum case name includes the enum name as a prefix (ignoring case
+and underscore characters), that prefix is stripped.
+If the stripped name would conflict with another entry in the
+same enum, the conflicting cases will have their respective numeric values
+appended to ensure the results are unique.
+For example:
+```protobuf
+syntax = "proto3";
+enum TestEnum {
+    TEST_ENUM_FOO = 0;
+    TESTENUM_BAR = 1;
+    BAZ = 2;
+    BAR = -3;
+}
+```
+becomes
+```swift
+enum TestEnum {
+  case foo = 0
+  case bar_1 = 1
+  case baz = 2
+  case bar_n3 = -3 // 'n' for "negative"
+}
+```
+
+Note #1: As explained in the protobuf language guide,
```

Comment:
My $0.02, i'd liken't leave tis off, as it isn't really about our generated API, and we probably don't want to have to duplicate all the google/protobuf docs about good/bad proto design.